### PR TITLE
add comment count to get/:article_id object

### PR DESCRIPTION
**Pull request by pzadev**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -438,3 +438,26 @@ describe("GET /api/articles (topic query)", () => {
       });
   });
 });
+
+describe("GET :article_id comment_count", () => {
```
Comment (NCDanOlver) [inline __tests__/app.test.js:442]: When we're doing our describe description it's good practice to include the full endpoint! We might use `:article_id` as a parametric input on another endpoint!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -438,3 +438,26 @@ describe("GET /api/articles (topic query)", () => {
...
+  test("should return comment_count where comment(s) match article_id ", () => {
+    return request(app)
+      .get("/api/articles/2")
+      .expect(200)
+      .then(({ body }) => {
+        const { article } = body;
+        expect(article).toHaveProperty("comment_count");
+        expect(article.comment_count).toEqual("1");
```
Comment (NCDanOlver) [inline __tests__/app.test.js:450]: Really good only checking for the `comment_count` key here as that's all we're introducing! 2 things to consider:
- Is there a more appropriate matcher for primitives than toEqual?
- Do we typically expect a `count` to be a string? If you were to change this, you may find some other tests have an issue just to warn!

Comment (pzadev) [reply __tests__/app.test.js:450]: fortunately only two failed! Used CAST now, thanks!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -438,3 +438,26 @@ describe("GET /api/articles (topic query)", () => {
...
+  });
+  test("should return total comment_count for article_id matching 1 ", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        const { article } = body;
+        expect(article).toHaveProperty("comment_count");
+        expect(article.comment_count).toEqual("11");
+      });
```
Comment (NCDanOlver) [inline __tests__/app.test.js:461]: Interesting test! I can see why you've considered this regarding building up complexity, but as the comment_count is coming from the database, we could also consider that having the key at all is sufficient (which would be covered by your previous test) as the management of the comment count is being done by tested code within PSQL.

File: models.js [hunk 4]
```diff
@@ -52,7 +52,17 @@ exports.fetchArticles = (sort_by = "created_at", order = "DESC", topic) => {
...
-  const text = "SELECT * FROM articles WHERE article_id = $1";
+
+  const text = `
+  SELECT articles.article_id, articles.topic, articles.author, articles.title, articles.body, 
+  articles.created_at, articles.votes, articles.article_img_url,
+  COUNT(comments.comment_id) AS comment_count
+  FROM articles
+  JOIN comments ON articles.article_id = comments.article_id
+  WHERE articles.article_id = $1
+  GROUP BY articles.article_id
```
Comment (NCDanOlver) [inline models.js:63]: Nice one! 🥇

**Comment by NCDanOlver** [comment]
Nice job with this! 🚀 I see you've already had a think about the data types 🥇 Interesting one with the 2nd GET test as well - I think this might fall into the "personal preference" camp but the NC way is you don't necessarily need it as explained in the comment.

Don't forget to update the `endpoints.json`!

**Pull request merged**